service/src/main/java/bio/terra/landingzone/library/landingzones/management/ResourcesDeleteManager.java

```diff
@@ -181,8 +169,20 @@ private void handleResourceDeletion(
     }
   }
 
-  private boolean isBaseResource(ResourceToDelete resourceToDelete) {
-    return LandingZoneBaseResourceType.containsValueIgnoringCase(
-        resourceToDelete.resource().type());
+  private int GetDeleteOrder(ResourceToDelete resourceToDelete) {
+    if (resourceToDelete.resource().resourceType() == null) {
+      return 0;
+    }
+
+    switch (resourceToDelete.resource().resourceType().toLowerCase()) {
+      case "virtualnetworks":
+        return 1;
+      case "privatednszones":
+        return 1;
```

Comment:
Is it intentional that this has the same priority order as `virtualnetworks`?

Reply:
Yeah, originally this code did the deletions in two passes isBaseResource() = false and true. virtualnetworks and privatednszones were the two resources defined as base resources so they were deleted second after the other resources. The NSG needs to be deleted after the vnet though, so I had to add this to enforce more ordering.